component/loki/source/heroku/heroku.go

```diff
@@ -30,38 +30,13 @@ func init() {
 // Arguments holds values which are used to configure the loki.source.heroku
 // component.
 type Arguments struct {
-	HerokuListener       ListenerConfig      `river:"listener,block"`
+	Server               *lhttp.ServerConfig `river:",squash"`
 	Labels               map[string]string   `river:"labels,attr,optional"`
 	UseIncomingTimestamp bool                `river:"use_incoming_timestamp,attr,optional"`
 	ForwardTo            []loki.LogsReceiver `river:"forward_to,attr"`
 	RelabelRules         flow_relabel.Rules  `river:"relabel_rules,attr,optional"`
 }
 
-// ListenerConfig defines a heroku listener.
-type ListenerConfig struct {
-	ListenAddress string `river:"address,attr,optional"`
-	ListenPort    int    `river:"port,attr"`
```

Comment:
Yhea, I guess if we remove the `optional` in the river config tags it should make the field required. What do you think of making the http port required, and the grpc optional (since I think the usual case is to use HTTP)?

Reply:
As discussed [here](https://github.com/grafana/agent/pull/3581#discussion_r1179284020), using a sane default for both http and grpc ports, so it's easy for a user to not configure the server at all and get it working.